### PR TITLE
Added auth keys and new container setup

### PR DESCRIPTION
**Pull request by jaytaph**
- Added initial auth-key system. It allows to create and list keys from the bm-client
- Added bm-server side of auth-key system (pretty much a duplicate of the api-key system)
- Added new container system that allows for easier mocking when needed (old container could not do this properly)
- Updated the API/AUTH key systems so they are merged into a single system (avoid lots of duplication)

### CODE REVIEW

File: cmd/bm-server/handler/auth_key.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+// ListAuthKeys returns a list of all keys for the given account
+func ListAuthKeys(w http.ResponseWriter, req *http.Request) {
+	h, err := hash.NewFromHash(mux.Vars(req)["addr"])
+	if err != nil {
+		ErrorOut(w, http.StatusNotFound, accountNotFound)
+		return
+	}
+
+	// Store Auth key into persistent storage
```
Comment (acalatrava) [inline cmd/bm-server/handler/auth_key.go:86]: Change comment

File: internal/container/proof-of-work.go [hunk 2]
```diff
@@ -20,30 +20,37 @@
...
+			}
 
-	//If redis.host is set on the config file it will use redis instead of bolt
-	if config.Server.Redis.Host != "" {
-		opts := redis.Options{
-			Addr: config.Server.Redis.Host,
-			DB:   config.Server.Redis.Db,
+			proofService = storage.NewRedis(&opts)
+		} else {
+			proofService = storage.NewBolt(&config.Server.Bolt.DatabasePath)
```
Comment (acalatrava) [inline internal/container/proof-of-work.go:46]: we need to add
```
if config.Server.Bolt.DatabasePath == "" {
 	config.Server.Bolt.DatabasePath = os.TempDir()
 }
```

File: internal/container/subscription.go [hunk 3]
```diff
@@ -47,9 +46,13 @@ func GetSubscriptionRepo() subscription.Repository {
 			return
 		}
 
-		//If redis is not set then it will use BoltDB as default
+		// If redis is not set then it will use BoltDB as default
 		subscriptionRepository = subscription.NewBoltRepository(config.Server.Bolt.DatabasePath)
```
Comment (acalatrava) [inline internal/container/subscription.go:50]: we need to add
```
if config.Server.Bolt.DatabasePath == "" {
		 config.Server.Bolt.DatabasePath = os.TempDir()
 }
```

File: internal/container/ticket.go [hunk 4]
```diff
@@ -52,5 +50,9 @@ func GetTicketRepo() ticket.Repository {
 		ticketRepository = ticket.NewBoltRepository(config.Server.Bolt.DatabasePath)
```
Comment (acalatrava) [inline internal/container/ticket.go:50]:
```
if config.Server.Bolt.DatabasePath == "" {
		 config.Server.Bolt.DatabasePath = os.TempDir()
 }
```

**Review by acalatrava — APPROVED** [review]
Go for it

File: cmd/bm-server/handler/auth_key.go (hunk 1 above)
Comment (jaytaph) [reply cmd/bm-server/handler/auth_key.go:86]: not sure into what? :-)

**Pull request merged**